src/app/test-utils/entity-matchers.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { Entity } from "../core/entity/model/entity";
+
+const entityMatchers: jasmine.CustomMatcherFactories = {
+  toHaveId: (util) => {
+    return {
+      compare: (entity: Entity, id: string) => {
+        const result = { pass: false, message: "" };
+        if (entity.getId() === id) {
+          result.pass = true;
```

Comment:
Generally, the functions are missing the negative (`.not...`) error messages. This should definitely be added for all of them otherwise error messages become less readable than they currently are.

Reply:
See https://jasmine.github.io/tutorials/custom_matcher#section-Failure_Messages